source/MaterialXRender/TinyObjLoader.cpp

```diff
@@ -63,23 +62,35 @@ bool TinyObjLoader::load(const FilePath& filePath, MeshList& meshList)
     MeshFloatBuffer& tangents = tangentStream->getData();
     mesh->addStream(tangentStream);
 
-    positions.resize(vertexCount * 3);
-    normals.resize(vertexCount * 3);
-    texcoords.resize(vertexCount * 2);
-    tangents.resize(vertexCount * 3);
+    // Explode the geometry, since we may have unshared geometry
+    // in position, normal or uv
+    size_t totalIndexCount = 0;
+    for (const tinyobj::shape_t& shape : shapes)
+    {
+        totalIndexCount += shape.mesh.indices.size();
+    }
+    positions.resize(totalIndexCount * 3);
```

Comment:
Should these numbers be constants?

Reply:
For the count*3 ? Yes, I can make the change.